### PR TITLE
added new class structure

### PR DESCRIPTION
**Pull request by shantanu7p**

### CODE REVIEW

File: Tests/MetricUnitConverterTest.php [hunk 1]
```diff
@@ -0,0 +1,337 @@
+<?php
+namespace ConverterApp;
+
+class MetricUnitConverterTest extends \PHPUnit_Framework_TestCase 
+{
+
```
Comment (kalehrishi) [inline Tests/MetricUnitConverterTest.php:6]: Code style is not according to PSR2 coding standard.

File: Tests/MetricUnitConverterTest.php [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+  *@test
+	 */
+ public function it_can_validate_any_positive_number_for_KiloMeter_to_meter_conversion()
+ { 
+	 $converter  = new MetricUnitConverter();
+	 $meter      = new Meter();
+	 $kilometer  = new KiloMeter();
+	 $currentDistance = new Distance(1,$kilometer);
+	 $newDistance     = new Distance(1000,$meter);
+	 $result=$converter->converter($currentDistance,$newDistance);
```
Comment (kalehrishi) [inline Tests/MetricUnitConverterTest.php]: this should return object not integer.

File: Tests/MetricUnitConverterTest.php [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+	/**
+  *@test
+	 */
+ public function it_can_validate_any_positive_number_for_KiloMeter_to_meter_conversion()
+ { 
+	 $converter  = new MetricUnitConverter();
+	 $meter      = new Meter();
+	 $kilometer  = new KiloMeter();
+	 $currentDistance = new Distance(1,$kilometer);
+	 $newDistance     = new Distance(1000,$meter);
```
Comment (kalehrishi) [inline Tests/MetricUnitConverterTest.php]: Why there is value for new distance is set to 1000? It should be 0.

File: Tests/MetricUnitConverterTest.php [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+	*
+	*@test
+	*@expectedException InvalidArgumentException
+	*/
+  public function it_should_throw_InvalidArgumentException_for_string_argument_for_centimeter_to_meter_conversion()
+  {
+ 	  $converter  = new MetricUnitConverter();
+	  $meter      = new Meter();
+	  $centimeter  = new CentiMeter();
+	  $currentDistance = new Distance("abc",$centimeter);
```
Comment (kalehrishi) [inline Tests/MetricUnitConverterTest.php]: Is it throwing exception for conversion or while creating object?

File: app/Distance.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+namespace ConverterApp;
+
+class Distance extends Value
+{
+  public function __construct($value,$unit)
+  {
+     if($value<0)
+      {
+     	 throw new \InvalidArgumentException("Please enter valid argument.");
```
Comment (kalehrishi) [inline app/Distance.php]: Message should be as clear as possible. In this case you can say `Value can't be negative`

File: Tests/MetricUnitConverterTest.php [hunk 6]
```diff
@@ -0,0 +1,337 @@
+<?php
+namespace ConverterApp;
```
Comment (kalehrishi) [inline Tests/MetricUnitConverterTest.php:2]: Where are tests for other classes?

File: app/CentiMeter.php [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace ConverterApp;
+
+class CentiMeter extends Length
+{
+  public $power;
```
Comment (kalehrishi) [inline app/CentiMeter.php]: Data members can't be public.

File: app/MetricUnitConverter.php [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+namespace ConverterApp;
+
+class MetricUnitConverter extends Converter
+{
+public function converter($currentUnit,$newUnit)
```
Comment (kalehrishi) [inline app/MetricUnitConverter.php]: You are accepting value not unit. so name of input param should be according to that.

File: app/MetricUnitConverter.php [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  { 
+    $calculatedPower = $currentUnit->unit->power - $newUnit->unit->power;
```
Comment (kalehrishi) [inline app/MetricUnitConverter.php]: Members of class shouldn't be accessed directly. Use getters and setters.

File: app/MetricUnitConverter.php (hunk 8 above)
Comment (kalehrishi) [inline app/MetricUnitConverter.php]: This class allows us to convert values with Metric units. So there should be checking for that in constructor. For that 
We should inherit all metric units from same base class `Metric` which will be inherited from `Unit` class. In constructor you can check if unit of both values passed is of Type `Metric`

`Length` class is not much useful in current scenario.

File: app/MetricUnitConverter.php~ [hunk 10]
```diff
@@ -0,0 +1,14 @@
+<?php
```
Comment (kalehrishi) [inline app/MetricUnitConverter.php~:1]: Why this file is committed?

File: app/Meter.php [hunk 11]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (kalehrishi) [inline app/Meter.php:1]: Why indentation is 2 spaces? It should be 4 spaces as per PSR2 coding standard.

File: app/KiloMeter.php [hunk 12]
```diff
@@ -0,0 +1,13 @@
+<?php
+
```
Comment (kalehrishi) [inline app/KiloMeter.php:2]: You should add comments as per php docs. 
http://www.phpdoc.org/

File: Tests/MetricUnitConverterTest.php (hunk 1 above)
Comment (shantanu7p) [inline Tests/MetricUnitConverterTest.php:6]: sir its according to psr4 standard

**Pull request closed without merging**